env0/data_template.go

```diff
@@ -131,6 +141,11 @@ func dataTemplateRead(ctx context.Context, d *schema.ResourceData, meta interfac
 		d.Set("github_installation_id", template.GithubInstallationId)
 	}
 
+	if template.TokenId != "" {
+		d.Set("token_id", template.TokenId)
+		d.Set("gitlab_project_id", template.GitlabProjectId)
+	}
```

Comment:
here also, we dont need to set GitlabProjectId if it is just Other VCS.
so probably need to do something like this:

```go
if template.TokenId != "" {
	d.Set("token_id", template.TokenId)
        if template.GitlabProjectId != 0 {
        	d.Set("gitlab_project_id", template.GitlabProjectId)
        }
}
```

Reply:
decc5707

Reply:
why outside the first `if`?
we should not Set it if there is no tokenID first

Reply:
it doesn't really matter in my taste,
WDYT?

Reply:
52220487
as we talked added to resource